packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, Prop, h, Listen} from '@stencil/core';
+import {buildInteractiveResult, Result, SmartSnippet} from '@coveo/headless';
+import {LinkWithResultAnalytics} from '../result-link/result-link';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {ResultContextEvent} from '../result-template-components/result-template-decorators';
+
+@Component({
+  tag: 'atomic-smart-snippet-source',
+  shadow: false,
+})
+export class AtomicSmartSnippetSource implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Prop() source!: Result;
+  @Prop() smartSnippet: SmartSnippet | undefined;
+  @Prop() isSuggestion!: boolean;
+  public error!: Error;
+
+  @Listen('atomic/resolveResult')
+  public resolveResult(event: ResultContextEvent<Result>) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (this.source) event.detail(this.source);
+  }
+
+  render() {
+    const interactiveResult = buildInteractiveResult(this.bindings.engine!, {
+      options: {result: this.source},
+    });
+
+    if (this.smartSnippet && !this.isSuggestion) {
+      return (
+        <section aria-label={this.bindings.i18n.t('smart-snippet-source')}>
+          <div part="source-url">
```

Comment:
I would remove the wrapping divs (`div[part="source-url"]` and `div[part="source-title"]`), and I would add their parts to their respective `LinkWithResultAnalytics` (similarly to what `atomic-smart-snippet-suggestions` does).